solver/src/solver/http_solver/model.rs

```diff
@@ -1,20 +1,23 @@
-use model::u256_decimal;
+use bigdecimal::BigDecimal;
+use ethcontract::H160;
+use model::u256_decimal::{self, DecimalU256};
 use primitive_types::U256;
 use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
 use std::collections::HashMap;
 
 #[derive(Debug, Default, Serialize)]
 pub struct BatchAuctionModel {
-    pub tokens: HashMap<String, TokenInfoModel>,
-    pub orders: HashMap<String, OrderModel>,
-    pub uniswaps: HashMap<String, UniswapModel>,
+    pub tokens: HashMap<H160, TokenInfoModel>,
+    pub orders: HashMap<usize, OrderModel>,
+    pub amms: HashMap<usize, PoolModel>,
```

Comment:
@twalth3r  - this is the place where "uniswaps" was changed to amms. Each member of this HashMap is a form of pool. We could use the name "pools" here instead although, like I mentioned, at some point this may include 0x and then the term pools would no longer be valid. We will probably first introduce a `StablePool` Model as well.

However given that this hasmap consists only of `PoolModels` and they all have a `pool_type` field I think maybe its best to go with `pools` instead of `amms`. Is there any other part of the code where you see an inconsistency?

Reply:
I was under the impression that you are not using the term `pool` at all in your code and was surprised to see `PoolModel` etc (was probably expecting `AmmModel`). But I guess you have some structural reasoning for when you use AMM and when Pool.

Reply:
In the JSON, if we call the field `amms`, then we should probably go with `amm_type` inside, or conversely for `pools`. Or maybe just `type`!?

Reply:
I think maybe we should just go with pools since I think the term PoolModels will be long lived. When the 0x integration we can cross that bridge later.

Reply:
I still think `pool.pool_type` is a bit redundant and slightly prefer `pool.type` or `pool.kind` if you want to get Rust-y (because `type` is a reserved keyword).

Reply:
Kind it is!